### DISCUSSION TITLE
Different dps_val for Button

### DISCUSSION THREAD

**Discussion by Flegm** [make-all/tuya-local#2267] [Q&A]
**Describe the bug**
In custom_components/tuya_local/devices/README.md:
![image](https://github.com/user-attachments/assets/defd2dbd-273b-466a-b490-2148be250c40)
but in Button.py:
`sync def async_press(self):
        """Press the button"""
        await self._button_dp.async_set_value(self._device, True)`

**To Reproduce**
Steps to reproduce the behavior including the device the issue was observed with:
Click on the Button send True always.

**Expected behavior**
I expect add mapping or default value with type different from boolean, and one send to my dp id.

![image](https://github.com/user-attachments/assets/23a79d3d-cedb-4c93-8eef-9b386980be39)
Command = True, but I add mapping with string command.

**Comment by make-all** [comment]
I'm not sure exactly what your problem is, but maybe [an example](https://github.com/make-all/tuya-local/blob/main/custom_components/tuya_local/devices/avidsen_soria_solarinverter.yaml#L45) will help.

**Reply by Peck07** [reply]
I think I've stumbled on the same...

I'm trying to use a Parkside PMRDA mower, but none of the buttons work, like the Start Fixed Mowing:
https://github.com/make-all/tuya-local/blob/main/custom_components/tuya_local/devices/moebot_s_mower.yaml#L89
This supposed to send out a command like 115, "StartFixedMowing", but instead sends 115, True:
(MainThread) [custom_components.tuya_local.device] Lawnmower sending dps update: {"115": "True"}

**Reply by make-all** [reply]
What are you doing to get to this point? It seems like you are sending the string "True" to the button somehow, instead of using the button press action.

**Reply by Peck07** [reply]
Nothing extraordinary, I'm just pressing the "Start fixed mowing" at the Controls (or on the dashboard) on the UI.

I think that's what Flegm tried to say, that it seems the button press sends always True here:
https://github.com/make-all/tuya-local/blob/main/custom_components/tuya_local/button.py#L60

**Reply by make-all** [reply]
No it doesn't, the function async_set_value maps the boolean True according to the config file, and I have just added a unit test to verify that it is correctly doing that, which passes locally (unfortunately pypi.org seems to be having a moment, which prevents the tests from running on github servers right now).

**Reply by Peck07** [reply]
I don't know exactly what happened, but now the buttons are working just fine... I have a theory, but that's not entirely valid as well :D
I realized I changed the local key, but the device id changed too... and tuya-local tried to send the commands with the wrong devid.
I deleted the device, recreated it, and the commands worked with a new "select" I've made (and pretty sure I've tried the existing buttons too, and didn't work), but now the existing buttons are working as well
